src/main/java/net/minecraftforge/items/itemhandlerobserver/IItemHandlerObserver.java

```diff
@@ -0,0 +1,17 @@
+package net.minecraftforge.items.itemhandlerobserver;
+
+import net.minecraft.item.ItemStack;
+import net.minecraftforge.items.IItemHandler;
+
+import javax.annotation.Nonnull;
+
+public interface IItemHandlerObserver
+{
+    void onStackInserted(IItemHandler handler, int slot, @Nonnull ItemStack newStack);
+
+    void onStackExtracted(IItemHandler handler, int slot, @Nonnull ItemStack newStack);
+
+    void onObserverableInvalidated();
+
+    boolean isNotValid();
```

Comment:
`isNot` usually sounds somewhat strange, especially when having to invert it and deal with double negatives. `isValid` should be enough.

Also choosing another synonyms might be an idea to avoid confusion about it. As this is about the observer still being valid, while the `onObserverableInvalidated` is about the observed inventory.